core/Pos/Core/Delegation.hs

```diff
@@ -0,0 +1,56 @@
+-- | Delegation payload types.
+
+module Pos.Core.Delegation
```

Comment:
I believe this module should also contains some stuff from `Pos.Core.Types`, namely:
```
       , ProxySigLight
       , ProxySKLight
       , ProxySigHeavy
       , ProxySKHeavy
```

Reply:
Somnevaus'

```
    Module imports form a cycle:
             module ‘Pos.Crypto.Signing.Tag’ (./Pos/Crypto/Signing/Tag.hs)
            imports ‘Pos.Core.Configuration.Protocol’ (./Pos/Core/Configuration/Protocol.hs)
      which imports ‘Pos.Core.Genesis.Types’ (./Pos/Core/Genesis/Types.hs)
      which imports ‘Pos.Core.Delegation’ (./Pos/Core/Delegation.hs)
      which imports ‘Pos.Core.Configuration’ (./Pos/Core/Configuration.hs)
      which imports ‘Pos.Core.Genesis.Canonical’ (./Pos/Core/Genesis/Canonical.hs)
      which imports ‘Pos.Core.Address’ (./Pos/Core/Address.hs)
      which imports ‘Pos.Crypto.Signing’ (./Pos/Crypto/Signing.hs)
      which imports ‘Pos.Crypto.Signing.Tag’ (./Pos/Crypto/Signing/Tag.hs)
```

Reply:
But it seems you don't need full `HasConfiguration` here, I believe you need only a small part (so you can import only `Pos.Core.Configuration.SmallPart`) which might resolve the problem.